### PR TITLE
codelldb: init at 1.10.0

### PR DESCRIPTION
**Pull request by Coosis**
## Description of changes

The package contains a binary of codelldb, used for DAP protocol for debugging purposes. This was packaged in a similar way as Mason(A neovim package manager) did. This is solely for DAP protocol, for advanced usages, check out [vscode-extensions.vadimcn.vscode-lldb](https://search.nixos.org/packages?channel=24.05&show=vscode-extensions.vadimcn.vscode-lldb&from=0&size=50&sort=relevance&type=packages&query=vscode-lldb).

## Things done
Install the 1.10.0 version of codelldb, unzip to expose the binary, create a symlink for accessing the binary.
Homepage is at https://github.com/vadimcn/codelldb.

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/co/codelldb/default.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      url = "https://github.com/vadimcn/codelldb/releases/download/v1.10.0/codelldb-aarch64-darwin.vsix";
+      sha256 = "0hbs2rr4r8zlii3srbc9xbmn5wm3p88cdsx85xp2vibbf9d7kc2a";
+    } else {
+      url = "";
+      sha256 = "";
+    }
+  else {
+    url = "";
+    sha256 = "";
+  };
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/codelldb/default.nix]: This must use an attrset instead of if else chains.

File: pkgs/by-name/co/codelldb/default.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{}:
+
+let
+  pkgs = import <nixpkgs> {};
+  stdenv = pkgs.stdenv;
+  lib = pkgs.lib;
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/codelldb/default.nix]:
```suggestion
```

File: pkgs/by-name/co/codelldb/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  else {
+    url = "";
+    sha256 = "";
+  };
+in
+pkgs.stdenv.mkDerivation {
+  pname = "codelldb";
+  version = "1.10.0";
+
+  nativeBuildInputs = [ pkgs.unzip ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/codelldb/default.nix]:
```suggestion
  nativeBuildInputs = [ unzip ];
```

File: pkgs/by-name/co/codelldb/default.nix [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  src = pkgs.fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/codelldb/default.nix]:
```suggestion
  src = fetchurl {
```

File: pkgs/by-name/co/codelldb/default.nix [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    url = binary.url;
+    sha256 = binary.sha256;
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/codelldb/default.nix]:
```suggestion
    inherit (binary) url hash;
```

File: pkgs/by-name/co/codelldb/default.nix [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+  unpackPhase = ''
+    mkdir -p $out/bin
+    unzip $src -d $out/bin
+  '';
+
+  installPhase = ''
+    ln -s $out/bin/extension/adapter/codelldb $out/bin/codelldb
```
Comment (SuperSandro2000) [inline pkgs/by-name/co/codelldb/default.nix]: The bin directory should not contain subdirectories

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Coosis** [comment]
@SuperSandro2000 Some problems were already addressed by later commits, please check them, others I have changed the .nix accordingly, please let me know if more changes are needed

**Comment by Coosis** [comment]
visit `https://github.com/Coosis/coosis-nix` if you really need codelldb.

**Pull request closed without merging**